packages/consensus/src/types/utreexo.cairo

```diff
@@ -111,6 +112,49 @@ pub impl UtreexoAccumulatorImpl of UtreexoAccumulator {
         self.num_leaves += 1_u64;
     }
 
+    /// Removes an output from the accumulator.
+    /// https://eprint.iacr.org/2019/611.pdf Algorithm 2 DeleteOne
+    fn delete(ref self: UtreexoState, proof: @UtreexoProof) {
+        let mut roots: Array<Option<felt252>> = array![];
+        let mut n: Option<felt252> = Option::None;
+        let mut h: usize = 0;
+
+        while h != (*proof.proof).len() {
+            let p = proof.proof[h];
+
+            if n != Option::None {
+                n = Option::Some(parent_hash(*p, n.unwrap()));
+                if h < self.roots.len() {
```

Comment:
Let's calculate the roots len once outside the loop

Reply:
done